packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -726,15 +553,29 @@ class GrpcRouteSpec extends RouteSpec {
     }
   }
 
-  public bind(_scope: Construct): RouteSpecConfig {
+  public bind(scope: Construct): RouteSpecConfig {
+    const serviceName = this.match.serviceName;
+    const methodName = this.match.methodName;
+    const metadata = this.match.metadata;
+
+    validateGprcMatch(this.match);
+    validateMetadata(metadata);
+
+    if (methodName && !serviceName) {
+      throw new Error('If you specify a method name, you must also specify a serviceName.');
```

Comment:
Either use the property name in both places, or in neither:

```suggestion
      throw new Error('If you specify a method name, you must also specify a service name');
```

Reply:
Thanks for the pointer!